app/src/main/java/org/mozilla/fenix/home/HomeMenu.kt

```diff
@@ -224,17 +410,19 @@ class HomeMenu(
             context.components.backgroundServices.accountManager.register(object : AccountObserver {
                 override fun onAuthenticationProblems() {
                     lifecycleOwner.lifecycleScope.launch(Dispatchers.Main) {
-                        onMenuBuilderChanged(BrowserMenuBuilder(
-                            listOf(reconnectToSyncItem) + coreMenuItems
-                        ))
+                        onMenuBuilderChanged(
```

Comment:
Could we define `menuItemsWithReconnectItem` within this block instead since it's only used here? Alternatively I could just imagine we can inline the if statement block in `BrowserMenuBuilder`.